pytorch_lightning/plugins/precision/ipu.py

```diff
@@ -27,9 +27,20 @@
 
 
 class IPUPrecisionPlugin(PrecisionPlugin):
-    """Precision plugin for IPU integration."""
+    """Precision plugin for IPU integration.
+
+    Raises:
+        MisconfigurationException:
+            If the precision is neither 16 nor 32.
+    """
 
     def __init__(self, precision: int) -> None:
+        supported_precision_values = (16, 32)
+        if precision not in supported_precision_values:
+            raise ValueError(
+                f"`Trainer(accelerator='ipu', precision={precision!r})` is not supported."
+                f" `precision` must be one of: {supported_precision_values}."
```

Comment:
So the reason we had the check in the accelerator connector is because the error message is not accurate when instantiating the plugin (or a subclass of it) outside the Trainer. So yeah, not ideal to move it here :)

Reply:
@awaelchli do you mean that user may init the precision plugin by themselves? Are you suggesting keep this in accl_connt?

Reply:
> do you mean that user may init the precision plugin by themselves? 

in general yes, that's what the plugins are meant to be used for. For the IPU precision specifically, nobody will. Still, the error message making a reference to the Trainer is misleading.